core/commands/swarm.go

```diff
@@ -447,6 +463,52 @@ func (ci connInfos) Swap(i, j int) {
 	ci.Peers[i], ci.Peers[j] = ci.Peers[j], ci.Peers[i]
 }
 
+func (ci *connInfo) identifyPeer(ps pstore.Peerstore, p peer.ID) (IdOutput, error) {
```

Comment:
I had to return a pointer to the IdOutput object, so that I could return nil,err in case some step fails

Reply:
do `return IdOutput{}, err` then :)